### PR TITLE
Il2cpp on mono - string tokens

### PR DESCRIPTION
**Pull request by ghost**
The new effort to run IL2CPP on the mono runtime instead of libil2cpp requires the metadata tokens for ldstr instructions, which Cecil currently discards.  This change stores these tokens in a dictionary on the MethodBody object, indexed by the instruction.

### CODE REVIEW

File: Mono.Cecil.Cil/MethodBody.cs [hunk 1]
```diff
@@ -77,7 +79,7 @@ public sealed class MethodBody : IVariableDefinitionProvider {
 			get { return scope; }
 			set { scope = value; }
 		}
-
+
```
Comment (mrvoorhe) [inline Mono.Cecil.Cil/MethodBody.cs]: Looks like an unnecessary spacing change.  If we are going to upstream this PR, they may appreciate removing it.

**Review by joshpeterson — APPROVED** [review]
LGTM

File: Mono.Cecil.Cil/MethodBody.cs [hunk 2]
```diff
@@ -106,12 +108,25 @@ static ParameterDefinition CreateThisParameter (MethodDefinition method)
...
 		}
 
 		public ILProcessor GetILProcessor ()
 		{
 			return new ILProcessor (this);
 		}
+
+		public bool GetInstructionToken (Instruction instruction, out MetadataToken token)
+		{
+			if (!instructionTokens.ContainsKey (instruction))
```
Comment (joncham) [inline Mono.Cecil.Cil/MethodBody.cs]: I think you could use 
`return instructionTokens .TryGetValue(instruction, token);`

**Comment by mrvoorhe** [comment]
lgtm

File: Mono.Cecil.Cil/MethodBody.cs (hunk 1 above)
Comment (joncham) [inline Mono.Cecil.Cil/MethodBody.cs]: Looks like extra whitespace?

File: Mono.Cecil.Cil/CodeReader.cs [hunk 3]
```diff
@@ -189,7 +189,9 @@ object ReadOperand (Instruction instruction)
 			case OperandType.InlineSig:
 				return GetCallSite (ReadToken ());
 			case OperandType.InlineString:
-				return GetString (ReadToken ());
+				var token = ReadToken ();
+				body.instructionTokens.Add (instruction, token);
+				return GetString (token);
```
Comment (joncham) [inline Mono.Cecil.Cil/CodeReader.cs:194]: Does this handle all string literals?

**Comment by TautvydasZilys** [comment]
Looks good to me.

**Comment by joncham** [comment]
LGTM

**Pull request merged**